### PR TITLE
Estimates as Data Ranges

### LINKED ISSUE #1627: Estimates as Data Ranges
Migrate from `estimate.year`, `estimate.q`  to `estimate.start` and `estimate.end`, where `start` and `end` are dates and goals should be filtered between them

### LINKED ISSUE #1669: Can't set estimate to 2024
The select works but then it's set to 2023

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Refactor
- [x] Feature

## Related issues

Resolve #1627
Resolve #1669

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/9020793/022aa8fc-9f7b-48a1-b8ee-1efaaf14eaf8

### CODE REVIEW

**Comment by sab94a** [comment]
Uploading Запись экрана 2023-09-09 в 17.58.56.mov…

**Comment by sab94a** [comment]
https://github.com/taskany-inc/issues/assets/9020793/f0830c60-93aa-41c0-9657-1e4091ad32fd

**Comment by sab94a** [comment]
Before Migration

telegram-cloud-photo-size-2-5188671830938995917-y
telegram-cloud-photo-size-2-5188671830938995919-y
telegram-cloud-photo-size-2-5188671830938995920-y

**Comment by sab94a** [comment]
Successful Migration!

`npx prisma migrate dev`
`npm run db:migrate:estimates`

https://github.com/taskany-inc/issues/assets/9020793/de49fea0-c5f8-4680-92f5-c4ff12e27263

**Comment by sab94a** [comment]
Redirects from legacy estimate filter: 

https://github.com/taskany-inc/issues/assets/9020793/f3d0bd2b-85bf-4ffc-9386-c37f2a8dea54

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "dev:db:run": "docker-compose -f docker-compose.dev.yml up -d",
     "dev:db:down": "docker-compose down",
     "dev:db:fill": "ts-node -r dotenv/config scripts/fillDevData.ts",
+    "db:migrate:estimates": "ts-node -r dotenv/config scripts/estimatesMigrate.ts",
```
Comment (awinogradov) [inline package.json:15]: What?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -208,7 +214,9 @@ model Goal {
   priority                String?
   priorityExact           Priority?             @relation(fields: [priorityId], references: [id])
   priorityId              Int?
-  estimate                EstimateToGoal[]
+  estimate                DateTime?
+  estimateType            DateType?
+  legacyEstimate          EstimateToGoal[]
```
Comment (awinogradov) [inline prisma/schema.prisma:219]: I think it must be vise versa. We don't need to rename current fields. Add new only.

File: scripts/estimatesMigrate.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Estimate } from '@prisma/client';
+
+import { convertLegasyEstimate, convertLegasyEstimateToHistory } from '../src/utils/dateTime';
+import { prisma } from '../src/utils/prisma';
+
+// TODO: Remove script here https://github.com/taskany-inc/issues/issues/1656
```
Comment (awinogradov) [inline scripts/estimatesMigrate.ts:6]: It doesn't help to understand what is this and why

File: src/components/GoalDependencyList/GoalDependencyList.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import styled from 'styled-components';
 import { Text, nullable, Table } from '@taskany/bricks';
 import { backgroundColor, danger0, gray4, gray9, textColor } from '@taskany/colors';
 import { IconBinOutline, IconMessageTextAltOutline } from '@taskany/icons';
-import { Estimate, State } from '@prisma/client';
+import { State, DateType } from '@prisma/client';
```
Comment (awinogradov) [inline src/components/GoalDependencyList/GoalDependencyList.tsx:6]: Sorry, we have issue about clean up such as imports

File: src/components/GoalEditForm/GoalEditForm.tsx [hunk 5]
```diff
@@ -22,22 +20,30 @@ interface GoalEditFormProps {
...
-        if (form.estimate && form.estimate.date) {
-            form.estimate.date = parseLocaleDate(form.estimate?.date, { locale }).toISOString();
-        }
-
         const updatedGoal = await update(form);
 
         onSubmit(updatedGoal);
     };
 
+    const { estimate, estimateType } = goal;
```
Comment (awinogradov) [inline src/components/GoalEditForm/GoalEditForm.tsx:34]: It helps really?

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Comment by asabotovich** [comment]
resolve #1669

**Pull request merged**